great_expectations/self_check/util.py

```diff
@@ -3065,15 +3064,15 @@ def check_json_test_result(  # noqa: C901 - 52
                 assert result["result"]["observed_value"] in value
 
             elif key == "unexpected_index_list":
-                if isinstance(data_asset, (SqlAlchemyDataset, SparkDFDataset)):
+                if isinstance(data_asset, SqlAlchemyDataset):
                     pass
-                elif isinstance(data_asset, (SqlAlchemyBatchData, SparkDFBatchData)):
+                elif isinstance(data_asset, SqlAlchemyBatchData):
```

Comment:
This enables Spark ID/PK tests for all Expectations defined as JSON files under `test_definitions`

Reply:
This is also the reason why 2 tests had to be adjusted below. They were basically not being tested before (at least not at the `unexpected_index_list`-level